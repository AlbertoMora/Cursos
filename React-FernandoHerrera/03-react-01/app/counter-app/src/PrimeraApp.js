import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitulo }) => {
    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    );
};

//Sirven para definir validaciones a las props
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
};

//Se puede hacer de esta forma, o estableciendo el valor por defecto directamente en la desestructuraci√≥n
//La diferencia es que de esta forma llega como una propiedad en el objeto props y de la otra forma no.
PrimeraApp.defaultProps = {
    subtitulo: "Soy un subtitulo"
};

export default PrimeraApp;
