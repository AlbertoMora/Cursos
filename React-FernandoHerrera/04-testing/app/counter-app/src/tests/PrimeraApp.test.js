import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import React from "react";
import { shallow } from "enzyme";

describe("Testing React components", () => {
    test('should show <PrimeraApp /> correctly"', () => {
        const saludo = "Hola bro";
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });
    test("should shown the subtitle send by props", () => {
        
        const saludo = "Hola bro";
        const subtitulo = "Este es un subt√≠tulo";
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);

    });

});
