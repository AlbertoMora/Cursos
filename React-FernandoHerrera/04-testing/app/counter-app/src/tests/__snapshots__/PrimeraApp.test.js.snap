// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing React components should show <PrimeraApp /> correctly" 1`] = `
<Fragment>
  <h1>
    Hola bro
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
